scrapy/spider.py

```diff
@@ -31,11 +31,19 @@ def __init__(self, name=None, **kwargs):
         if not hasattr(self, 'start_urls'):
             self.start_urls = []
 
-    def log(self, message, level=log.DEBUG, **kw):
-        """Log the given messages at the given log level. Always use this
-        method to send log messages from your spider
+    @property
+    def logger(self):
+        logger = logging.getLogger(self.name)
```

Comment:
@chekunkov It shouldn't break existing spiders while running locally, all log options are applied to the root logger so they're going to be propagated to all loggers, not just scrapy descendants.

Code in hworkers has to adapt to support all these changes (remove the manual twisted logging handling), and it should support using any python loggers too. I'm not sure about this but I think it already supports python logging, but it automatically sets its level to INFO: https://github.com/scrapinghub/hworker/blob/master/hworker/bot/log.py#L40